### PR TITLE
Constants package

### LINKED ISSUE #882: Make constants globally available
Constants such as the Docker labels need to be globally available across all packages.
Currently the constants are buried in respective packages that are hard to export without introducing circular dependencies.

### PR DESCRIPTION
**Pull request by adambkaplan**
Moved docker labels and s2i scripts constants to separate top-level package.
Fixes #882

### CODE REVIEW

File: pkg/constants/labels.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+	// BuildSourceContextDir is the Docker image LABEL that S2I uses to record the context directory in the source code repository to use for the  build.
+	//
+	// During a rebuild, this label is used by S2I to set the context directory within the source code for the S2I build.
+	BuildSourceContextDirLabel = buildNamespace + "source-context-dir"
+
+	// TODO: Deprecate BuilderVersionLabel?
+
+	// BuilderBaseVersionLabel is the Docker image LABEL that tells S2I the version of the base image used by the builder image.
+	BuilderBaseVersionLabel = OpenshiftNamespace + "builder-base-version"
```
Comment (adambkaplan) [inline pkg/constants/labels.go]: I noticed that this label and `BuilderVersionLabel` weren't really used by s2i.

**Comment by adambkaplan** [comment]
[test]

**Comment by bparees** [comment]
in general we keep constants that are exposed to users in the api package to reflect that they are part of the api (we can't change their values w/o potentially breaking someone).

File: pkg/docker/docker.go [hunk 2]
```diff
@@ -44,24 +45,6 @@ const (
 	// DestinationLabel instead.
 	LocationEnvironment = "STI_LOCATION"
```
[the remark this answers is not in the document]
Comment (bparees) [reply pkg/docker/docker.go]: yes, might as well, as long as we still have code referencing the values.

**Comment by adambkaplan** [comment]
@bparees fixed test failures & squashed.

File: pkg/api/constants/scripts.go [hunk 3]
```diff
@@ -7,32 +7,40 @@ import (
...
-	// depending on the builder image used.
+	// Environment contains list of key value pairs that will be set during the STI build.
+	// Users can use this file to provide extra configuration depending on the builder image used.
 	Environment = "environment"
 )
 
 const (
 	// UserScripts is the location of scripts downloaded from user provided URL (-s flag).
 	UserScripts = "downloads" + string(os.PathSeparator) + "scripts"
+
```
[the remark this answers is not in the document]
Comment (adambkaplan) [reply pkg/api/constants/scripts.go]: think the first one is the name of the s2i scripts, second are the paths where the scripts are uploaded/downloaded

[the remark this answers is not in the document]
Comment (bparees) [reply pkg/api/constants/scripts.go]: yeah i don't think there's any need to keep it as two blocks

**Comment by adambkaplan** [comment]
@bparees rebased & squashed

**Comment by bparees** [comment]
this is going to be blocked until https://github.com/openshift-s2i/s2i-wildfly/pull/169 merges and a new wildfly image is published.

**Comment by bparees** [comment]
[merge]

**Comment by bparees** [comment]
now waiting on this fix :(
https://github.com/openshift-s2i/s2i-wildfly/pull/170

**Comment by bparees** [comment]
[merge]

**Pull request merged**